### PR TITLE
Enum for Card Types

### LINKED ISSUE #90: Implement card types as enums
enums > strings

### PR DESCRIPTION
**Pull request by jaypatel1227**
Added a CardTye Enum and replaced all functionality that relied on string card types. Note that since cardType is not yet a module that is a part of pyminion, the examples in bots/examples/ don't work as they can't find pyminion.cardType.

Closes #90.

### CODE REVIEW

File: pyminion/game.py [hunk 1]
```diff
@@ -115,9 +116,9 @@ def _create_basic_piles(self) -> List[Pile]:
                 basic_piles.append(Pile([card] * SILVER_LENGTH))
             elif card.name == "Gold":
                 basic_piles.append(Pile([card] * GOLD_LENGTH))
-            elif "Victory" in card.type:
+            elif CardType.Victory in card.type:
                 basic_piles.append(Pile([card] * VICTORY_LENGTH))
-            elif card.name == "Curse":
```
Comment (evanofslack) [inline pyminion/game.py:120]: This as well should remain a string. Or better yet, copy the above victory card check. 

```python
elif CardType.Curse in card.type:
                basic_piles.append(Pile([card] * CURSE_LENGTH))
```

**Review by evanofslack — CHANGES_REQUESTED** [review]
@jaypatel1227 Thanks for the contribution. I really appreciate you reading through the issues and making this change. I think Enums are much better than just having Strings thrown around.

Looks like all the tests passed, which is great. I just made some comments about some nit picky things. If you can update to address those points it should be good to merge in.

File: pyminion/cardtype.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+"""
```
[the remark this answers is not in the document]
Comment (jaypatel1227) [reply pyminion/cardtype.py:1]: moved to the top of core since you use card type in the definition of various Card classes

**Review by jaypatel1227 — COMMENTED** [review]
Made the changes you requested. Got rid of the cardtype file and rolled it into core.

**Review by evanofslack — APPROVED** [review]

**Comment by evanofslack** [comment]
Fantastic, thanks @jaypatel1227

**Pull request merged**